### ISSUE TITLE
neural network

### ISSUE THREAD

**Issue by rrwu23** [rrwu23/Chessie#1]
I need to create a neural network only with 3 layers which can recongnize patterns such as opposition, fork, pin, sweker, and discovered checks

`List<float> weights` will be weights
`List<float> biases` will be biases
```
     public float ReLu(float val){
             if (val < 0) {return 0;}
             return val
     }
```
simple implementation of ReLu
values **a0** to **an**
weights **w0** to **wn**
newval_k = ReLu(a0w0 + a1w1 + ... anwn - bias)

**Comment by rrwu23** [comment]
hi

**Comment by eajajhossain** [comment]
Hi @rrwu23 I noticed this issue has been in progress for about a month  Are you still actively working on it? If not, then  I’d be happy to help implement the 3-layer network with ReLU activation and pattern classification.
My proposed approach would be:
1.Define fixed-size input encoding for board state
2.Implement 3-layer with feedforward network 
3.Use ReLU activation
4.Output logits for each tactical pattern class

Let me know if you'd like assistance or if there are specific constraints I should be aware of.
